frontend/src/views/WheresMyGene/components/GeneSearchBar/components/SaveImage/index.tsx

```diff
@@ -114,28 +155,40 @@ export default function SaveImage({
      */
     const download = debounce(
       download_({
-        fileType,
+        allChartProps,
+        availableFilters,
+        availableOrganisms,
         heatmapNode,
         observer,
         selectedCellTypes,
+        selectedFileTypes,
+        selectedFilters,
         selectedGenes,
+        selectedOrganismId,
         selectedTissues,
         setEchartsRendererMode,
-        setIsDownloading,
+        setDownloadStatus,
       }),
       MUTATION_OBSERVER_TIMEOUT
     );
 
     observer.observe(heatmapNode, config);
 
-    setEchartsRendererMode("svg");
+    if (selectedFileTypes.includes("svg")) {
+      setEchartsRendererMode("svg");
+    }
```

Comment:
because the svg echarts rendering mode is only required for svg, for CSV and PNG it can stay as canvas. This prevents any unnecessary blinking from rerendering back to canvas from svg